src/HDF5.jl

```diff
@@ -1087,8 +1087,11 @@ dataspace(sz1::Int, sz2::Int, sz3::Int...; max_dims::Union{Dims,Tuple{}}=()) = _
 
 
 function get_dims(dspace::Dataspace)
-    dims = h5s_get_simple_extent_dims(dspace)
-    return tuple(reverse!(dims[1])...), tuple(reverse!(dims[2])...)
+    N, h5_dims, h5_maxdims = h5s_get_simple_extent_dims(dspace)
+    # reverse dimensions since hdf5 uses C-style order
+    dims = ntuple(i -> h5_dims[N-i+1] .% Int, N)
```

Comment:
https://support.hdfgroup.org/HDF5/faq/limits.html#:~:text=There%20are%20no%20limitations%20on%20the%20file%20size.,datatypes%20or%20compound%20datatypes%20that%20are%20very%20large 

Makes me  uncomfortableabout the truncation

Reply:
I guess the face that `dataspace` only accepts `Dims{N}`, makes me more comfortable that this is fine. Albiet, there might me some edge case with esoteric HDF5 files with really large dimensions don't work with HDF5.jl. (probably something else to look into, because I'm assuming there would be more than this as the only issue).

Reply:
Yeah, I haven't been worrying about it because it _seems_ like all of Julia's array constructors cast all sizes to `Int` before calling into the C runtime. I suspect there would be fundamental issues with working with arrays larger than `typemax(Int)` axes, so it's not likely this would be the single point of failure.